libs/ledger-live-common/src/bridge/impl.ts

```diff
@@ -85,5 +97,60 @@ export function getAccountBridgeByFamily(family: string, accountId?: string): Ac
   if (!jsBridge) {
     throw new CurrencyNotSupported("account bridge not found " + family);
   }
-  return jsBridge.accountBridge;
+  return wrapAccountBridge(jsBridge.accountBridge);
+}
+
+function wrapAccountBridge<T extends TransactionCommon>(
+  bridge: AccountBridge<T>,
+): AccountBridge<T> {
+  return {
+    ...bridge,
+    getTransactionStatus: async (...args) => {
+      const blockchainTransactionStatus = await bridge.getTransactionStatus(...args);
+
+      const account = args[0];
+      if (!isCheckSanctionedAddressEnabled(account.currency)) {
+        return blockchainTransactionStatus;
+      }
+
+      const commonTransactionStatus = await commonGetTransactionStatus(...args);
+      return mergeResults(blockchainTransactionStatus, commonTransactionStatus);
+    },
+  };
+}
+
+function mergeResults(
+  blockchainTransactionStatus: TransactionStatusCommon,
+  commonTransactionStatus: Partial<TransactionStatusCommon>,
+): TransactionStatusCommon {
+  const errors = { ...blockchainTransactionStatus.errors, ...commonTransactionStatus.errors };
+  const warnings = { ...blockchainTransactionStatus.warnings, ...commonTransactionStatus.warnings };
+  return { ...blockchainTransactionStatus, errors, warnings };
+}
+
+async function commonGetTransactionStatus(
+  account: Account,
+  transaction: TransactionCommon,
+): Promise<Partial<TransactionStatusCommon>> {
+  const errors: Record<string, Error> = {};
+  const warnings: Record<string, Error> = {};
+
+  let isRecipientSanctioned = false;
+  if (transaction.recipient && transaction.recipient !== "") {
+    isRecipientSanctioned = await isAddressSanctioned(account.currency, transaction.recipient);
+    if (isRecipientSanctioned) {
+      errors.recipient = new AddressesSanctionedError("AddressesSanctionedError", {
+        addresses: [transaction.recipient],
+      });
```

Comment:
All this code should probably live in some other file

This change seems like it will remove flexibility by adding this layer of indirection

We could probably had created a helper fonction like `makeAccountBridgeReceive` used in every coin from the framework